openapi.yaml

```diff
@@ -586,51 +586,44 @@ parameters:
-    'invalid_filter_field' respectively. This is accompanied by the 'codeMajor/severity' values
-    of 'failure/error'."
+  '401':
+    description: "The request was not correctly authorised i.e. 'unauthorisedrequest."
     schema:
       $ref: '#/definitions/ErrorResult'
-  UnprocessableEntity:
-    description: "This error condition may occur if a JSON request body contains well-formed (i.e.
-    syntactically correct), but semantically erroneous, JSON instructions. This would be accompanied
-    by the 'codeMajor/severity' values of 'failure/error'."
+  '403':
+    description: "This is used to indicate that the server can be reached and process the request
+      but refuses to take any further action i.e. 'forbidden'."
     schema:
       $ref: '#/definitions/ErrorResult'
-  TooManyRequests:
-    description: "The server is receiving too many requests i.e. 'server_busy'. Retry at a later
-    time. This would be accompanied by the 'codeMajor/severity' values of 'failure/error'."
+  '404':
+    description: "Either the supplied identifier is unknown in the Service Provider and so the
+      object could not be changed or an invalid GUID has been supplied. The statement 'Unknown
+      Object' should also be presented."
     schema:
       $ref: '#/definitions/ErrorResult'
-  Unauthorized:
-    description: "The request was not correctly authorised i.e. 'unauthorisedrequest'. This would
-    be accompanied by the 'codeMajor/severity' values of 'failure/error'."
+  '422':
+    description: "This error condition may occur if a JSON request body contains well-formed (i.e.
+      syntactically correct), but semantically erroneous, JSON instructions."
     schema:
       $ref: '#/definitions/ErrorResult'
-  Forbidden:
-    description: "This is used to indicate that the server can be reached and process the request
-    but refuses to take any further action i.e. 'forbidden'. This would be accompanied by the
-    'codeMajor/severity' values of 'failure/error'."
+  '429':
+    description: "The server is receiving too many requests i.e. 'server_busy'. Retry at a later
+      time."
     schema:
       $ref: '#/definitions/ErrorResult'
-  NotFound:
-    description: "Either the supplied identifier is unknown in the Service Provider and so the
-    object could not be changed or an invalid GUID has been supplied. This would be accompanied by
-    the 'codeMajor/severity' values of 'failure/error'. The statement 'Unknown Object' should
-    also be presented."
+  '500':
+    description: "This code should be used only if there is catastrophic error and there is not a
+      more appropriate code i.e. 'internal_server_error'."
     schema:
       $ref: '#/definitions/ErrorResult'
-  Default:
-    description: "This is the default error response when no other appropriate code is available.
-    This would be accompanied by the 'codeMajor/severity' values of 'failure/error'."
+  default:
```

Comment:
What is the point of having a default error object? Given that their description of this object says that it would be accompanied by 'failure/error', and that value is shared for their 400 responses and not 500 responses, it seems like a general 400 error should be returned when "no other appropriate code is available"

Reply:
I agree that the default error shouldn't be necessary. From our discussion, I think we could replace this with a 500